app/utils.go

```diff
@@ -0,0 +1,131 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/kv"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+)
+
+// SetupSimulation creates the config, db (levelDB), temporary directory and logger for
+// the simulation tests. If `FlagEnabledValue` is false it skips the current test.
+// Returns error on an invalid db intantiation or temp dir creation.
+func SetupSimulation(dirPrefix, dbName string) (simtypes.Config, dbm.DB, string, log.Logger, bool, error) {
```

Comment:
in `master` we want to clearly separate what's in test what's not. For example this function shouldn't be part of the API.